### PR TITLE
Ensure that NameOnBankAccount on only appears for refunds

### PR DESCRIPTION
**Pull request by anisharamnani**
## Link to pivotal/JIRA issue
https://codeforamerica.atlassian.net/browse/FYST-1448
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
Ensure that NameOnBankAccount on only appears for refunds
Taxes Owed does not need bank account name

### CODE REVIEW

**Comment by powersurge360** [comment]
Can you leave a comment somewhere near the `payment_or_deposit_type` in the intake to mention that `direct_debit` doesn't literally exist and what constitutes it? That's where I intuitively went when I was looking for this and I think folks who are looking for it would do a grep on the code base and likely find it.

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb [hunk 1]
```diff
@@ -657,5 +659,16 @@
...
+    context "when there are taxes owed" do
+      before do
+        allow(intake).to receive(:refund_or_owe_taxes_type).and_return(:owe)
+        allow(intake).to receive(:payment_or_deposit_type).and_return(:direct_deposit)
+      end
+
+      it 'does not include the NameOnBankAccount attribute' do
+        expect(xml.at("Form502 NameOnBankAccount")).not_to be_present
+      end
+    end
```
Comment (arinchoi03) [inline spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb:672]: can we move this spec to the
```
    context "Line 51d: NameOnBankAccount" do
```
and test that joint account holder name also does not show up as well?

**Review by arinchoi03 — APPROVED** [review]
looks great! 🕺

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb (hunk 1 above)
Comment (anisharamnani) [reply spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb]: yes have done!

**Pull request merged**